pcli/src/command/tx.rs

```diff
@@ -808,6 +830,63 @@ impl TxCmd {
                     .await?;
                 app.build_and_submit_transaction(plan).await?;
             }
+            TxCmd::Ics20Withdrawal {
+                destination_chain_id,
+                destination_chain_address,
+                amount,
+                denom,
+                timeout_height,
+                timeout_timestamp,
+                source_channel,
+                return_address_index,
+            } => {
+                let fee = Fee::from_staking_token_amount(1_000_000u64.into());
+                let (ephemeral_return_address, _) = app
+                    .fvk
+                    .ephemeral_address(OsRng, AddressIndex::from(*return_address_index));
+                let account_group_id = app.fvk.account_group_id();
+                let current_height = app.view().status(account_group_id).await?.sync_height;
+
+                // get the current time on the local machine
+                let current_time_u64 = SystemTime::now()
+                    .duration_since(UNIX_EPOCH)
+                    .expect("Time went backwards")
+                    .as_secs();
+
+                let mut timeout_height = *timeout_height;
+                if timeout_height == 0u64 {
+                    // add two days to height, assuming 6 blocks per minute
+                    timeout_height = current_height + 28800u64;
+                }
+                let mut timeout_timestamp = *timeout_timestamp;
+                if timeout_timestamp == 0u64 {
+                    // add 2 days to current time
+                    timeout_timestamp = current_time_u64 + 172800u64;
+                }
+
+                let denom = asset::REGISTRY.parse_denom(denom).unwrap();
+                let amount = Amount::try_from(amount.clone()).unwrap();
+
+                let plan = Planner::new(OsRng)
+                    .ics20_withdrawal(
+                        destination_chain_id.clone(),
+                        destination_chain_address.clone(),
+                        denom,
+                        amount,
+                        timeout_height,
+                        timeout_timestamp,
+                        ephemeral_return_address,
+                        ChannelId::from_str(source_channel)?,
+                    )
+                    .fee(fee)
+                    .plan(
+                        app.view.as_mut().unwrap(),
+                        app.fvk.account_group_id(),
+                        AddressIndex::new(0),
```

Comment:
I think this should match the `source` (current `return_address_index`), so that we're always withdrawing from the specified account and returning funds to it.